sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/context/QuerySpecification.java

```diff
@@ -117,13 +125,22 @@ public Void visitGroupByElement(GroupByElementContext ctx) {
 
     @Override
     public Void visitAggregateFunctionCall(AggregateFunctionCallContext ctx) {
-      aggregators.add((AggregateFunction) visitAstExpression(ctx));
+      aggregators.add(AstDSL.alias(getTextInQuery(ctx), visitAstExpression(ctx)));
       return super.visitAggregateFunctionCall(ctx);
     }
 
     private UnresolvedExpression visitAstExpression(ParseTree tree) {
       return expressionBuilder.visit(tree);
     }
+
+    /**
+     * Get original text in query.
+     */
+    private String getTextInQuery(ParserRuleContext ctx) {
+      Token start = ctx.getStart();
+      Token stop = ctx.getStop();
+      return queryString.substring(start.getStartIndex(), stop.getStopIndex() + 1);
+    }
```

Comment:
np: same as `AstBuilder.getTextInQuery()` and can move to some parsing util class?

Reply:
done.